src/graph/NodesCore.hpp

```diff
@@ -217,12 +218,14 @@ struct YieldPointsNode : IPointsNodeSingleInput
 	std::vector<rgl_field_t> getRequiredFieldList() const override { return fields; }
 
 	// Data getters
-	IAnyArray::ConstPtr getFieldData(rgl_field_t field) override
-	{ return results.at(field); }
+	IAnyArray::ConstPtr getFieldData(rgl_field_t field) override { return results.at(field); }
```

Comment:
`YieldPointsNode` has a missing override for `hasField(rgl_field_t field)`. The node queries a parent node for field existence while getting field data from its internal map.
Also, I have found that the map is not cleared when setting a new set of fields required. The user could receive an outdated point cloud in some cases.

I think we should discuss how `YieldPointsNode` should work:
1. Allow to retrieve only fields defined in this node.
2. To retrieve fields from this node and parent nodes.

Changes described in this comment could be done with another task/PR.

Reply:
> Also, I have found that the map is not cleared when setting a new set of fields required. The user could receive an outdated point cloud in some cases.

That's true - when requesting a field that was removed from the set of fields. Very specific scenario though.

> I think we should discuss how YieldPointsNode should work.

I think this is one of the problems (relatively minor) that stem from the lack of separation between data and operations (i.e. splitting point cloud out of nodes). I would prefer not to focus on such problems, and aim for fixing the core issue.

Reply:
I agree with you that we should aim to fix the root cause but since the temporary fixes are easy to apply we can make the yield node perform properly (without noticed bugs).